drivers/rtc/Kconfig

```diff
@@ -3,3 +3,9 @@
 
 config RTC
 	bool "RTC driver support"
+
+if RTC
+
+source "drivers/rtc/Kconfig.rtc_emul"
```

Comment:
From what I can tell, the convention for Kconfig fragments is to just keep the driver name and not the class. So maybe rename `Kconfig.rtc_emul` -> `Kconfig.emul`.

E.g.
```shell
ls drivers/ieee802154/Kconfig*
drivers/ieee802154/Kconfig
drivers/ieee802154/Kconfig.b91
drivers/ieee802154/Kconfig.cc1200
drivers/ieee802154/Kconfig.cc13xx_cc26xx
drivers/ieee802154/Kconfig.cc2520
drivers/ieee802154/Kconfig.dw1000
drivers/ieee802154/Kconfig.kw41z
drivers/ieee802154/Kconfig.mcr20a
drivers/ieee802154/Kconfig.nrf5
drivers/ieee802154/Kconfig.rf2xx
drivers/ieee802154/Kconfig.uart_pipe
```